iModelCore/ECDb/ECDb/SchemaSync.cpp

```diff
@@ -91,19 +76,31 @@ ProfileVersion SchemaSyncHelper::QueryProfileVersion(DbR db, ProfileKind kind) {
 // @bsimethod
 //+---------------+---------------+---------------+---------------+---------------+------
 DbResult SchemaSyncHelper::SaveProfileVersion(SchemaSync::SyncDbUri syncDbUri, ProfileKind kind, ProfileVersion const& ver) {
-    const auto profileSpec = SchemaSyncHelper::GetPropertySpec(kind);
-
     Db conn;
     Db::OpenParams openParams(Db::OpenMode::ReadWrite);
     SchemaSync::ParseQueryParams(openParams, syncDbUri);
     auto rc = conn.OpenBeSQLiteDb(syncDbUri.GetUri().c_str(), openParams);
     if (rc != BE_SQLITE_OK) {
         return rc;
     }
+
+    return SaveProfileVersion(conn, kind, ver);
+}
+//---------------------------------------------------------------------------------------
+// @bsimethod
+//+---------------+---------------+---------------+---------------+---------------+------
+DbResult SchemaSyncHelper::SaveProfileVersion(DbR conn, ProfileKind kind, ProfileVersion const& ver) {
+    const auto profileSpec = SchemaSyncHelper::GetPropertySpec(kind);
+    if (conn.IsReadonly()) {
+        LOG.error("Failed to save profile version. Database is readonly.");
+        return BE_SQLITE_ERROR;
+    }
     Utf8String profileVersionStr = ver.ToJson();
-    rc = conn.SavePropertyString(profileSpec, profileVersionStr);
-    if (rc != BE_SQLITE_OK)
+    auto rc = conn.SavePropertyString(profileSpec, profileVersionStr);
+    if (rc != BE_SQLITE_OK) {
+        LOG.error("Failed to save profile version.");
```

Comment:
Don't want to log `rc`?

Reply:
I added log error messages where i was just return rc.